src/snap.py

```diff
@@ -48,6 +48,8 @@ def install(self) -> bool:
             self.kafka = kafka
             self.kafka.connect(plug="removable-media")
 
+            snap.hold_refresh(forever=True)
```

Comment:
This is not something I'm particularly comfortable with at charm scope. This will prevent **any** snap refresh, including the core snaps which include critical security fixes.

We need to be careful to only hold the individual snaps that we're installing in each charm.

Reply:
Hm, that makes sense. In that case we _do_ need the per snap feature at `operator-libs-linux`

Reply:
The ideal flow here is to install a particular **revision** of the snap (i.e. `snap install kafka --revision 45`), and then hold the snap at that specified revision.

Reply:
I understand. But these are two different features to add to the `operator-libs-linux.snap` lib no?
1) Needed - hold per snap
2) Great to have, but not blocking 1 - specify revision number on install